source/Octopus.Tentacle/Kubernetes/KubernetesModule.cs

```diff
@@ -42,6 +42,9 @@ protected override void Load(ContainerBuilder builder)
             builder.Register<KubernetesAgentMetrics>(ctx => 
                 ctx.Resolve<KubernetesAgentMetrics.Factory>()
                     .Invoke(ctx.ResolveNamed<IPersistenceProvider>(kubernetesAgentMetricsPersistence)));
+            
+            builder.RegisterType<KubernetesEventMonitorTask>().As<IBackgroundTask>().SingleInstance();
```

Comment:
Because this is registered in the `KubernetesModule` it will only run with the Kubernetes agent which is what we want. 👍

Reply:
I literally just worked that out :D